upf/walkers/simplifier.py

```diff
@@ -36,9 +38,20 @@ def _number_to_fnode(self, value: Union[int, float, Fraction]) -> FNode:
             fnode = self.manager.Real(Fraction(value))
         return fnode
 
-    def simplify(self, expression: FNode) -> FNode:
-        """Performs basic simplification of the given expression."""
-        return self.walk(expression)
+    def simplify(self, expression: FNode, problem: Optional['upf.model.Problem'] = None) -> FNode:
+        """Performs basic simplification of the given expression.
+
+        If a problem is given, it also uses the static fluents of the problem and removes quantifiers
+        for a better simplification."""
+        if problem is not None:
+            self.static_fluents = problem.get_static_fluents()
+            expression_without_quantifiers = self.quantifiers_remover.remove_quantifiers(expression, problem)
```

Comment:
This can be costly: why not leaving the user the freedom of doing it manually with the QuantifiersRemover transformer?